### DISCUSSION TITLE
Proper Way to Upgrade Deskflow on Mac

### DISCUSSION THREAD

**Discussion by Dentrax** [deskflow/deskflow#7925] [Q&A]
I'd like to upgrade Deskflow on my Mac 15.1.1 from v1.17.0 to v1.17.2, but it keep saying the following error:

```
The operation can’t be completed because the item “Deskflow.app” is in use.
```

I've ensured that this app is not running at all. No process on the Activity Monitor.

```
ps auxwww | grep -i deskflow
```

Returns nothing.

Ensured there is no such `.plist` under `/Library/LaunchAgents/`.

Only `lsof` returns something but not sure if its related:

```
sudo lsof | grep -i deskflow
mds         551                  root   31r      DIR               1,25         306                   2 /Volumes/Deskflow
mds         551                  root   33r      DIR               1,25         306                   2 /Volumes/Deskflow
mds         551                  root   35r      DIR               1,25         306                   2 /Volumes/Deskflow
```

What else that I can check?

**Comment by sithlord48** [comment]
it looks like you have the DMG file mounted still

**Reply by Dentrax** [reply]
I tried to run those:

```
sudo mdutil -i off /Volumes/Deskflow # disable spotlight indexing
hdiutil detach /Volumes/Deskflow # eject corresponding disk
umount -f /Volumes/Deskflow # unmount
```

Now I can able to open that DMG file at all, probably restart is required or smth.

**Comment by sithlord48** [comment]
To complete this answer. If deskflow had crashed macOS thinks the server (or client ) is running and the os tries to lock the file. You can remove it with `rm -rf <pathTo>/Deskflow.app` . However until you reboot mac os will think the file is in use if you create it at the same path 

Example: 
 If the "inuse" Deskflow was `~/Desktop/Deskflow.app` and I remove it via the cli then run `touch ~/Desktop/Deskflow.app` . You will be unable to remove the empty file because it will be "inuse" this will happen until you reboot.  I do this alot when I develop on mac os the kill button in the IDE has the same effect since it just kills the Application.

**Comment by phillijw** [comment]
I just had the same exact issue today upgrading from 1.17 to 1.23. I ended up installing to a different folder.
